atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/knowledge/Bucket.java

```diff
@@ -17,19 +17,26 @@
 package com.palantir.atlasdb.transaction.knowledge;
 
 import com.palantir.atlasdb.AtlasDbConstants;
+import org.immutables.value.Value;
 
-public final class AbortedTimestampUtils {
-    private AbortedTimestampUtils() {}
+@Value.Immutable
+public interface Bucket {
+    @Value.Parameter
+    long value();
 
-    public static long getMaxTsInCurrentBucket(long bucket) {
-        return getMinTsInBucket(bucket + 1) - 1;
+    static Bucket of(long value) {
```

Comment:
nit: I might name this `ofIndex` or something like that, to be clear that no one accidentally tries to pass a timestamp in.